Library/Homebrew/development_tools.rb

```diff
@@ -79,6 +82,9 @@ def clang_build_version
         if (path = locate("clang")) &&
            build_version = `#{path} --version`[/clang-(\d{2,})/, 1]
           Version.new build_version
+        elsif (path = locate("clang")) &&
+          build_version = `#{path} --version`[/clang version (\d\.\d\.\d)/, 1]
+          Version.new build_version
```

Comment:
I'd suggest removing this chunk and changing line 83 to (untested)
```ruby
build_version = Utils.popen_read(path, "--version")[/clang(-| version )(\d{2,})/, 2]
```
If `clang --version` also displays `clang version x.x.x` on Ubuntu, this change can go upstream to Homebrew/brew.

Reply:
Seems to!

```
$ clang --version
clang version 4.0.1-6 (tags/RELEASE_401/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

$ which clang
/usr/bin/clang

$ dpkg -S /usr/bin/clang
clang: /usr/bin/clang

$ dpkg -l clang
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                 Version         Architecture    Description
+++-====================-===============-===============-=============================================
ii  clang                1:4.0-37~exp3ub amd64           C, C++ and Objective-C compiler (LLVM based)
```

Reply:
Same result under Ubuntu:

```
[2018-03-17 05:26:40 PM -0700 (Sat)]$ /usr/bin/clang --version
clang version 4.0.1-6 (tags/RELEASE_401/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
(git:brew android-linker) buck@tortle-2016:~/prefix/brew 
[2018-03-17 05:26:50 PM -0700 (Sat)]$ dpkg -S /usr/bin/clang  
clang: /usr/bin/clang
(git:brew android-linker) buck@tortle-2016:~/prefix/brew 
[2018-03-17 05:26:54 PM -0700 (Sat)]$ dpkg -l clang           
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                 Version         Architecture    Description
+++-====================-===============-===============-==============================================
ii  clang                1:4.0-37~exp3ub amd64           C, C++ and Objective-C compiler (LLVM based)
(git:brew android-linker) buck@tortle-2016:~/prefix/brew 
[2018-03-17 05:26:57 PM -0700 (Sat)]$ lsb_release -a          
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 17.10
Release:	17.10
Codename:	artful
```

Reply:
https://github.com/Homebrew/brew/pull/3943